collect/collect.go

```diff
@@ -789,7 +796,12 @@ func (i *InMemCollector) processSpan(ctx context.Context, sp *types.Span, source
 	var spanForwarded bool
 	// if this trace doesn't belong to us and it's not in sent state, we should forward a decision span to its decider
 	if !trace.Sent && !isMyTrace {
-		i.Metrics.Increment(source + "_router_peer")
+		switch source {
+		case "peer":
```

Comment:
Why is source still a string? Shouldn't you also use the enum here?

Reply:
It was because of circular dependency. I just moved the RouterType into the `types` package and updated the code here to reuse that